File: fix_video.dart
import 'dart:io';

void main() {
  final scratch = File(r'C:\Users\Priyajit Bhowmik\.gemini\antigravity\brain\7e2e15eb-23c2-4912-b232-ed42e5836f5e\scratch\videographer_dashboard.dart.txt').readAsStringSync();
  
  final fStart1 = scratch.indexOf('    Widget _buildFinanceTab');
  final fStart = scratch.indexOf('    Widget _buildFinanceTab', fStart1 + 1);
  final fEnd = scratch.indexOf('  Widget _buildProfileTab');
  final financeTab = scratch.substring(fStart, fEnd);

  final targetFile = File(r'lib\screens\videographer_dashboard.dart');
  final target = targetFile.readAsStringSync();

  final startIdx = target.indexOf('  Widget _buildSessionApprovalCard');
  final endIdx = target.indexOf('  void _showEditPersonalDetailsDialog');

  final sessionCard = '''  Widget _buildSessionApprovalCard(BuildContext context, AppState state, Task t) {
    final client = state.clients.where((c) => c.id == t.clientId).firstOrNull;
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 6),
      child: Row(
        children: [
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(client?.name ?? 'Unknown', style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 12)),
                Text('\ | ?\',
                    style: const TextStyle(fontSize: 11, color: Colors.black54)),
              ],
            ),
          ),
          ElevatedButton(
            style: ElevatedButton.styleFrom(
              backgroundColor: SageColors.primary,
              foregroundColor: Colors.white,
              padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 8),
              textStyle: const TextStyle(fontSize: 10, fontWeight: FontWeight.bold),
            ),
            onPressed: () => context.read<AppState>().approveVideographerSession(t.id),
            child: const Text('APPROVE'),
          ),
        ],
      ),
    );
  }

''';

  final profileTab = '''  Widget _buildProfileTab(BuildContext context, AppState state, Persona persona) {
    final emp = state.employees.firstWhere((e) => e.id == persona.id);
    final isVideoEditor = emp.role.toLowerCase().contains('video editor');
    final unpaidSessionsList = state.tasks.where((t) => t.assignedTo == emp.id && (isVideoEditor ? true : t.taskType == 'Session') && t.isCompleted && !t.isPaidToVideographer).toList();
    final int unpaidSessionsCount = unpaidSessionsList.length;
    
    double pendingPayout = 0;
    for (var t in unpaidSessionsList) {
      if (isVideoEditor) {
        pendingPayout += emp.perSessionRate;
      } else {
        final c = state.clients.where((client) => client.id == t.clientId).firstOrNull;
        if (c != null) {
          pendingPayout += c.sessionRate;
        }
      }
    }

    final displayPayout = emp.paymentCleared ? emp.pendingPayAmount : pendingPayout;
    final displaySessions = emp.paymentCleared ? (emp.pendingPayMonth ?? "\") : "\";

    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      children: [
        const SizedBox(height: 10),
        TerminalPanel(
          title: "ACCOUNT DATA",
          child: Column(
            children: [
              Center(
                  child: ClipOval(
                    child: Image.asset(availableAvatars[emp.avatar % availableAvatars.length], fit: BoxFit.cover, width: 140, height: 140),
                  ),
                ),
              const SizedBox(height: 16),
              _profileRow("NAME", emp.name),
              _profileRow("ROLE", persona.roleLabel),
              _profileRow("ID CODE", persona.id),
              _profileRow("PASSWORD", emp.password),
              _profileRow("ADDRESS", emp.address.isNotEmpty ? emp.address : '---'),
              _profileRow("PHONE", emp.phone.isNotEmpty ? emp.phone : '---'),
              _profileRow("EMAIL", emp.email.isNotEmpty ? emp.email : '---'),
              const SizedBox(height: 24),
            ],
          ),
        ),
        
        const SizedBox(height: 16),
        TerminalPanel(
          title: "FINANCE DATA",
          child: Column(
            children: [
              _profileRow("PENDING PAYOUT", "?\"),
              _profileRow("UNPAID SESSIONS", displaySessions),
              const SizedBox(height: 16),
              SizedBox(
                width: double.infinity,
                child: ElevatedButton(
                  onPressed: emp.paymentCleared
                      ? () {
                          context.read<AppState>().toggleEmployeePaymentApproved(emp.id, true);
                          ScaffoldMessenger.of(context).showSnackBar(
                            const SnackBar(content: Text("Payment Receipt Confirmed!", style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold))),
                          );
                        }
                      : null,
                  style: ElevatedButton.styleFrom(
                    backgroundColor: emp.paymentCleared ? SageColors.primary : Colors.grey.shade300,
                    foregroundColor: emp.paymentCleared ? Colors.white : Colors.grey.shade600,
                  ),
                  child: Text(emp.paymentCleared ? "RECEIVE PAYMENT" : "WAITING FOR PAYMENT"),
                ),
              ),
            ],
          ),
        ),
      ],
    );
  }

''';

  final newContent = target.substring(0, startIdx) + sessionCard + financeTab + profileTab + target.substring(endIdx);
  targetFile.writeAsStringSync(newContent);
  print('Success');
}

File: fix_video.js
const fs=require('fs');
const p='C:/Users/Priyajit Bhowmik/Downloads/n sage os/sage os/sage os/sage_mainframe/lib/screens/videographer_dashboard.dart';
let t=fs.readFileSync(p,'utf8');
const orig=fs.readFileSync('C:/Users/Priyajit Bhowmik/Downloads/n sage os/sage os/sage os/sage_mainframe/temp_cal_video.txt','utf8');

const replacement = `Widget _buildCalendar(AppState state, Persona persona) {
    final now = DateTime.now();
    final mySessionTasks = state.tasks.where((t) => t.assignedTo == persona.id && t.taskType == 'Session').toList();

    return SageCalendar(
      currentMonth: _calendarMonth,
      onPreviousMonth: () => setState(() => _calendarMonth = DateTime(_calendarMonth.year, _calendarMonth.month - 1, 1)),
      onNextMonth: () => setState(() => _calendarMonth = DateTime(_calendarMonth.year, _calendarMonth.month + 1, 1)),
      legend: Row(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          calendarLegend(SageColors.primary, 'Completed'),
          const SizedBox(width: 14),
          calendarLegend(SageColors.tertiary, 'Booked'),
          const SizedBox(width: 14),
          calendarLegend(SageColors.error, 'Pending'),
        ],
      ),
      cellBuilder: (ctx, date) {
        final sessions = mySessionTasks.where((t) {
          final dd = t.deadline;
          return dd.year == date.year && dd.month == date.month && dd.day == date.day;
        }).toList();
        final hasBooked = sessions.any((t) => t.isApprovedByVideographer && !t.isCompleted);
        final hasPending = sessions.any((t) => !t.isApprovedByVideographer && !t.isCompleted);
        final hasCompleted = sessions.any((t) => t.isCompleted);
        final isToday = date.year == now.year && date.month == now.month && date.day == now.day;
        final isSel = _selectedDate?.year == date.year && _selectedDate?.month == date.month && _selectedDate?.day == date.day;

        Color bgColor = Colors.transparent;
        if (hasCompleted) bgColor = SageColors.primary;
        else if (hasBooked) bgColor = SageColors.tertiary;
        else if (hasPending) bgColor = SageColors.error;
        else if (isToday) bgColor = SageColors.yellowAccent;

        if (isSel && (bgColor == Colors.transparent || bgColor == SageColors.yellowAccent)) {
          bgColor = Colors.black;
        }

        Color textColor = (bgColor == SageColors.primary || bgColor == SageColors.tertiary || bgColor == SageColors.error || bgColor == Colors.black) ? Colors.white : Colors.black87;

        return GestureDetector(
          onTap: () => setState(() => _selectedDate = date),
          child: Container(
            margin: const EdgeInsets.all(2),
            decoration: BoxDecoration(
              color: bgColor,
              borderRadius: BorderRadius.circular(8),
              border: Border.all(color: Colors.black, width: isSel ? 2.5 : 1.0),
            ),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Text('\${date.day}', style: TextStyle(fontSize: 11, fontWeight: FontWeight.bold, color: textColor)),
              ],
            ),
          ),
        );
      },
    );
  }

  `;

t = t.replace(orig, replacement);

const legendStart = t.indexOf('Widget _legend(Color c, String label)');
if (legendStart !== -1) {
  const legendEnd = t.indexOf(']);', legendStart) + 3;
  t = t.substring(0, legendStart) + t.substring(legendEnd);
}

fs.writeFileSync(p, t);
console.log('Video dashboard updated');
